### DISCUSSION TITLE
Custom icon for the SearchBar Component

### DISCUSSION THREAD

**Discussion by KevinvdBurg** [software-mansion/react-native-screens#1955] [Ideas]
In developing our iOS application using React Native, we are looking to enhance the user experience by incorporating a custom search icon within the headerSearchBarOptions in React Navigation. While native iOS development supports this customization, we have observed that the react-native-screens library does not currently expose an API to change the default search icon to a custom image.

**References**

[How to change UISearchBar Icon to a custom image - Stack Overflow](https://stackoverflow.com/questions/37805754/how-to-change-uisearchbar-icon-to-custom-image)

**Feature Description**

We propose the following feature for react-native-screens:

An API that allows React Native developers to set a custom search icon in the UISearchBar.
This API should be flexible enough to accept local asset references or external URIs for the custom icon image.

**Use Case**

A custom search icon is often a critical part of an app's brand identity and user interface design. By enabling this feature, react-native-screens would allow developers to maintain brand consistency and improve the aesthetic alignment of their apps with their brand guidelines.

**Comment by tboba** [comment]
Hi @KevinvdBurg, thanks for creating this thread! 😄

I believe we've already thought about something like that, for sure I believe there was a thread that suggests to add support for changing the color of search bar icon 🔎 to darker one when using the dark theme, but this is probably not related to setting custom icon.

We'll consider this one!
